Modules/_zoneinfo.c

```diff
@@ -2213,7 +2216,7 @@ find_ttinfo(zoneinfo_state *state, PyZoneInfo_ZoneInfo *self, PyObject *dt)
     }
     else if (!num_trans || ts > local_transitions[self->num_transitions - 1]) {
         return find_tzrule_ttinfo(&(self->tzrule_after), ts, fold,
-                                  PyDateTime_GET_YEAR(dt));
+                                  PyDateTime_GET_YEAR(dt_dt));
```

Comment:
Need to deallocate `dt_dt` before returning here, and similarly in all other return paths.